AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -785,6 +785,33 @@ - (NSMutableDictionary *)getReportingParametersForUpdatedRelease:(NSString *)upd
   return reportingParameters;
 }
 
+- (void)changeDistributionGroupIdAfterAppUpdateIfNeeded {
+  NSString *lastDownloadedReleaseHashes = [MS_USER_DEFAULTS objectForKey:kMSDownloadedReleaseHashKey];
+  NSString *lastDownloadedDistributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDownloadedDistributionGroupIdKey];
+  if ((lastDownloadedDistributionGroupId == nil) || ([self isCurrentReleaseWasUpdated:lastDownloadedReleaseHashes] == NO)) {
+    return;
+  }
+  NSString *currentDistributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDistributionGroupIdKey];
+  if ((currentDistributionGroupId != nil) && ([lastDownloadedDistributionGroupId isEqualToString:currentDistributionGroupId])) {
+    return;
+  }
+
+  // Set group ID from downloaded release details.
+  MSLogDebug([MSDistribute logTag], @"Current group ID doesn't match the group ID of downloaded release, updating current group id: %@", lastDownloadedDistributionGroupId);
+  [MS_USER_DEFAULTS setObject:lastDownloadedDistributionGroupId forKey:kMSDistributionGroupIdKey];
```

Comment:
Why do you set `setObject:lastDownloadedDistributionGroupId` here instead of `currentDistributionGroupId`?

Reply:
One user can be added to different distribution groups and the user may download and update an release from another group. So after update and restart, we need to use another distribution group ID and update it in storage.

I added some code comments so I hope it should be clearer.